### PR TITLE
Support to open pull request to another repo

### PR DESCRIPTION
**Pull request by mszhanyi**
I need to submit PRs to another PR but GITHUB_REPOSITORY always refers to current repo.
So I make this little change, maybe others need this option as well

### CODE REVIEW

**Comment by vsoch** [comment]
Thanks @mszhanyi that's an interesting idea! I should be able to take a look over the weekend.

File: pull-request.py [hunk 1]
```diff
@@ -268,14 +269,17 @@ def add_reviewers(entry, reviewers, team_reviewers):
 API_VERSION = "v3"
 BASE = "https://api.github.com"
 
+pr_token = os.environ["PULL_REQUEST_TOKEN"] if "PULL_REQUEST_TOKEN" in os.environ else get_envar("GITHUB_TOKEN")
```
Comment (vsoch) [inline pull-request.py:272]: Just taking a quick look before the weekend - this does look reasonable, but could you please update:

1. the README to include all new variables and descriptions
2. add a full example to the examples folder of doing a pr for a different repository

And these lines can be simplified to be:

```suggestion
PR_TOKEN = os.environ.get("PULL_REQUEST_TOKEN") or get_envar("GITHUB_TOKEN")
```
And I made all caps to be consistent with other variables around it.

File: pull-request.py [hunk 2]
```diff
@@ -268,14 +269,17 @@ def add_reviewers(entry, reviewers, team_reviewers):
...
+pr_repo = os.environ["PULL_REQUEST_REPOSITORY"] if "PULL_REQUEST_REPOSITORY" in os.environ else get_envar("GITHUB_REPOSITORY")
```
Comment (vsoch) [inline pull-request.py:273]:
```suggestion
PR_REPO = os.environ.get("PULL_REQUEST_REPOSITORY") or get_envar("GITHUB_REPOSITORY")
```

File: pull-request.py [hunk 3]
```diff
@@ -268,14 +269,17 @@ def add_reviewers(entry, reviewers, team_reviewers):
...
+
 HEADERS = {
-    "Authorization": "token %s" % get_envar("GITHUB_TOKEN"),
+    "Authorization": "token %s" % pr_token,
```
Comment (vsoch) [inline pull-request.py:276]:
```suggestion
    "Authorization": "token %s" % PR_TOKEN,
```

File: pull-request.py [hunk 4]
```diff
@@ -268,14 +269,17 @@ def add_reviewers(entry, reviewers, team_reviewers):
...
     "Accept": "application/vnd.github.%s+json;application/vnd.github.antiope-preview+json;application/vnd.github.shadow-cat-preview+json"
     % API_VERSION,
 }
 
 # URLs
-REPO_URL = "%s/repos/%s" % (BASE, get_envar("GITHUB_REPOSITORY"))
+REPO_URL = "%s/repos/%s" % (BASE, pr_repo)
```
Comment (vsoch) [inline pull-request.py:282]:
```suggestion
REPO_URL = "%s/repos/%s" % (BASE, PR_REPO)
```

File: pull-request.py (hunk 1 above)
Comment (mszhanyi) [reply pull-request.py]: thanks, fixed

File: README.md [hunk 5]
```diff
@@ -39,6 +39,8 @@ Unlike standard actions, this action just uses variables from the environment.
 | Name | Description | Required | Default |
 |------|-------------|----------|---------|
 | BRANCH_PREFIX | the prefix to filter to. If the branch doesn't start with the prefix, it will be ignored | false | "" |
+| PULL_REQUEST_REPOSITORY | Choose another repository instead of default GITHUB_REPOSITORY for the PR  | false | |
+| PULL_REQUEST_TOKEN | [Personal Access Token(PAT)](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token) for PULL_REQUEST_REPOSITORY | false | |
```
Comment (vsoch) [inline README.md:43]:
```suggestion
| PULL_REQUEST_TOKEN | [Personal Access Token(PAT)](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token) only if you define a different repository with PULL_REQUEST_REPOSITORY | false | |
```

File: README.md [hunk 6]
```diff
@@ -71,6 +73,8 @@ an issue or PR, they are ignored otherwise.
 The `GITHUB_TOKEN` secret is required to interact and authenticate with the GitHub API to open
 the pull request. The example is [deployed here](https://github.com/vsoch/pull-request-action-example) with an example opened (and merged) [pull request here](https://github.com/vsoch/pull-request-action-example/pull/1) if needed.
```
Comment (vsoch) [inline README.md:74]: @mszhanyi I'm wondering if there is a way to make this less complicated. E.g., the user still could use the GITHUB_TOKEN variable defined in secrets to be a PAT? Are there cases when both would be wanted so this wouldn't work?

And the same question for `GITHUB_REPOSITORY`. Is there a way to not have two variables for the "same" thing?

Comment (mszhanyi) [reply README.md:74]: According to https://docs.github.com/en/actions/reference/environment-variables#naming-conventions-for-environment-variables , It looks it's not necessary to define GITHUB_XXX.

**Review by vsoch — APPROVED** [review]

**Pull request merged**